Mathlib/Geometry/Euclidean/Basic.lean

```diff
@@ -436,6 +436,22 @@ theorem dist_sq_eq_dist_orthogonalProjection_sq_add_dist_orthogonalProjection_sq
   exact Submodule.inner_right_of_mem_orthogonal (vsub_orthogonalProjection_mem_direction p₂ hp₁)
     (orthogonalProjection_vsub_mem_direction_orthogonal s p₂)
 
+/-- The distance between a point and its orthogonal projection to a subspace equals the distance
+to that subspace as given by `Metric.infDist`. This is not a `simp` lemma since the simplest form
+depends on the context (if any calculations are to be done with the distance, the version with
+the orthogonal projection gives access to more lemmas about orthogonal projections that may be
+useful). -/
+lemma dist_orthogonalProjection_eq_infDist (s : AffineSubspace ℝ P) [Nonempty s]
```

Comment:
Could you add the `nndist` version of this too, with `infNndist`? it might be easier to pove that one first.

Reply:
I've added it (proved second, not first, given the shortage of lemmas about `infNndist`).